ls1intum/Artemis
# `Assessment`: Fix an issue where deleting one unsaved inline feedback deletes all unsaved feedbacks

### Checklist
#### General



#### Client

### Motivation and Context


Closes #7709 

### Description

To correctly identify the feedback that should be deleted before it has been saved for the first time, the line, file and content of the feedback are now taken into consideration in addition to the id.

### Steps for Testing

Prerequisites:
- 1 Tutor/Instructor
- 1 Programming Exercise with manual assessment

1. As tutor/instructor assess a programming exercise.
2. Add a few inline comments in the code editor.
3. Delete one of those comments.
4. The other comments should still be there.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
